src/khepri_projection.erl

```diff
@@ -303,18 +303,28 @@ delete(#khepri_projection{name = Name}) ->
 trigger(
   #khepri_projection{name = Name, projection_fun = ProjectionFun},
   Path, OldProps, NewProps) ->
-    Table = ets:whereis(Name),
-    case ProjectionFun of
-        copy ->
-            trigger_copy_projection(Name, Table, OldProps, NewProps);
-        StandaloneFun ->
-            case ?HORUS_STANDALONE_FUN_ARITY(StandaloneFun) of
-                2 ->
-                    trigger_simple_projection(
-                      Table, Name, StandaloneFun, Path, OldProps, NewProps);
-                4 ->
-                    trigger_extended_projection(
-                      Table, Name, StandaloneFun, Path, OldProps, NewProps)
+    case ets:whereis(Name) of
```

Comment:
I have a question first, mostly for my curiosity: why did we resolve the table name first instead of passing it to ETS?

Reply:
And another question about the fix: which process executes this code?

In other words, can aux effects be executed concurrently?

I’m wondering if the ETS table can be deleted between that `ets:whereis/1` and the actual `ets:{insert,delete}()`. If aux effects are always executed sequentially for now, I think I would be more confortable if the "table doesn’t exist" error was caught by the `catch` later, to make the check atomic. This would be more future-proof as well if the design of aux effects changes some day.

Something like:
```erlang
try
    ets:insert(...)
catch
    Class:Reason:Stacktrace ->
        case ets:whereis(Name) of
            undefined ->
                %% Table doesn't exist
            _ ->
                %% Other error, we log
...
```

If concurrent execution of aux effects is implemented, the reverse is also true: the table could appear after that `ets:whereis/1` and before the insert/delete.

What do you think?

Reply:
Ok, thank you for the feedback!